py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,489 @@
+                                                           self.format_time_to)
+
+            self._check_warn_threshold(time_delta, event_str['start_time'],
+                                       event_str['end_time'], event_str['summary'])
+            self._check_button_open(index, url)
+
+            curr_event_format = self._check_button_toggle(index)
+
+            if self.force_lower_case:
+                event_str = dict((k, v.lower())
+                                 if v is not None else (k, v)
+                                 for k, v in event_str.items())
+
+            event_formatted = self.py3.safe_format(
+                curr_event_format,
+                {'summary': event_str['summary'],
+                 'location': event_str['location'],
+                 'description': event_str['description'],
+                 'start_time': event_str['start_time'],
+                 'end_time': event_str['end_time'],
+                 'start_date': event_str['start_date'],
+                 'end_date': event_str['end_date'],
+                 'time_to': time_to_composite})
+
+            self.py3.composite_update(event_formatted, {'index': index})
+            responses.append(event_formatted)
+
+            index += 1
+
+        format_separator = self.py3.safe_format(self.format_separator)
+        self.py3.composite_update(format_separator, {'index': 'sep'})
+        responses = self.py3.composite_join(format_separator, responses)
+
+        return {'events': responses}
+
+    def google_calendar(self):
+        """
+        The method that outputs the response.
+
+        First, we check credential authorization. If no authorization, we display an error message,
+        and try authorizing again in 5 seconds.
+
+        Otherwise, we fetch the events, build the response, and output the resulting composite.
+        """
+        if not self.is_authorized:
+            self.is_authorized = self._authorize_credentials()
+
+            return {'cached_until': self.py3.time_in(self.cache_timeout),
+                    'full_text': self.py3.safe_format(self.format)}
+        else:
+            if not self.no_update:
+                self.events = self._get_events()
+
+            return {
+                'cached_until': self.py3.time_in(self.cache_timeout),
+                'composite': self.py3.safe_format(self.format, self._build_response())
+            }
+
+    def on_click(self, event):
+        if self.is_authorized and self.events is not None:
+            """
+            If button_refresh is clicked, we allow the events to be updated if the
+            last event update occured at least 1 second ago. This prevents a bug that
+            can crash py3status since refreshing the module too fast results in
+            incomplete event information being fetched as _get_events() is called
+            repeatedly.
+
+            Otherwise, we disable event updates.
```

Comment:
I looked at your  `_check_button_open` and `_check_button_toggle`.

It may be easier if you make `self.url` and work with `on_click` to open them instead of using `self.no_update`, `self.button`, and whatnot that you have here to make things work. Maybe one of them just get tripped wrong?  See spaceapi and github for good examples... as this could be the issue you are struggling with.